### PR TITLE
composite checkout: Add EPS support

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* Add EPS payments support to composite checkout.

image

#### Testing instructions

- Use D45560-code to force this payment method to be available from the API
- Sandbox the store.
- Add a plan to your cart and visit checkout. 
- Add `?flags=composite-checkout-force` to the query string in the URL and reload the page because currently it is disabled for international currencies.
- Verify that "EPS e-Pay" is an option displayed in the final payment step.
- Verify that you cannot pay if the "Name" field is empty.
- Fill out the "Name" field and submit the payment.
- Verify that you are redirected to a Stripe test page.
- Click to accept the payment and verify that the payment completes successfully and that the plan is purchased.

### CODE REVIEW

**Comment by fditrapani** [comment]
The logo was looking a little aliased, I think because of the image tag, and the copy under the logo was undistinguishable so I replaced the image with SVG markup and removed the sub text too. Here's the updated version:

image

**Comment by sirbrillig** [comment]
@fditrapani that looks awesome, but did you maybe forget to push your changes? I don't see them on this branch.

**Comment by fditrapani** [comment]
I did forget :D Just pushed an update.

File: client/my-sites/checkout/composite-checkout/composite-checkout.js [hunk 1]
```diff
@@ -478,7 +478,9 @@ export default function CompositeCheckout( {
 			giropay: ( transactionData ) =>
 				genericRedirectProcessor( 'giropay', transactionData, getThankYouUrl, isWhiteGloveOffer ),
 			ideal: ( transactionData ) =>
-				genericRedirectProcessor( 'ideal', transactionData, getThankYouUrl, isWhiteGloveOffer ),
+							genericRedirectProcessor( 'ideal', transactionData, getThankYouUrl, isWhiteGloveOffer ),
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/composite-checkout.js:481]: Looks like this file could use a `prettier` run to fix the indentation.

File: client/my-sites/checkout/composite-checkout/payment-method-helpers.js [hunk 2]
```diff
@@ -125,6 +125,16 @@ export async function submitStripeRedirectTransaction( paymentMethodId, transact
 	return submit( formattedTransactionData );
 }
 
+export async function submitStripeEpsTransaction( transactionData, submit ) {
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/payment-method-helpers.js:128]: This is not necessary after the addition of `genericRedirectProcessor`

File: client/my-sites/checkout/composite-checkout/use-create-payment-methods.js [hunk 3]
```diff
@@ -284,6 +318,7 @@ export default function useCreatePaymentMethods( {
 		...existingCardMethods,
 		idealMethod,
 		giropayMethod,
+        epsMethod,
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/use-create-payment-methods.js:321]: Very minor indentation issue here.

File: packages/composite-checkout/src/lib/payment-methods/eps.js [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+const debug = debugFactory( 'composite-checkout:eps-payment-method' );
+
+export function createEpsPaymentMethodStore() {
+	debug( 'creating a new eps payment method store' );
+	const actions = {
+		changeCustomerName( payload ) {
+			return { type: 'CUSTOMER_NAME_SET', payload };
+		},
+		changeCustomerBank( payload ) {
```
Comment (sirbrillig) [inline packages/composite-checkout/src/lib/payment-methods/eps.js:35]: This store has two data fields: the name and the bank, but down below I only see an input for the name. I think we can remove all mentions of "bank" from this file.

File: packages/composite-checkout/src/lib/payment-methods/eps.js [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+	const customerName = useSelect( ( select ) => select( 'eps' ).getCustomerName() );
+
+	return (
+		<Button
+			disabled={ disabled }
+			onClick={ () => {
+				if ( isFormValid( store ) ) {
+					debug( 'submitting eps payment' );
+					setTransactionPending();
+					onEvent( { type: 'REDIRECT_TRANSACTION_BEGIN' } );
```
Comment (sirbrillig) [inline packages/composite-checkout/src/lib/payment-methods/eps.js:159]: This event requires a payload to specify the type, like this.
```suggestion
					onEvent( { type: 'REDIRECT_TRANSACTION_BEGIN', payload: { paymentMethodId: 'eps' } } );
```

**Comment by nsakaimbo** [comment]
The job linked in the wp-desktop review shows this build error:

```
$ check-npm-client && node ${NODE_ARGS:---max_old_space_size=8192} ./node_modules/webpack/bin/webpack.js --config client/webpack.config.js --display errors-only && yarn run build-languages-if-enabled
 
ERROR in ./my-sites/checkout/composite-checkout/composite-checkout.js
 
Module build failed (from ../node_modules/babel-loader/lib/index.js):
 
SyntaxError: /home/circleci/wp-calypso/client/my-sites/checkout/composite-checkout/composite-checkout.js: Unexpected token (481:1)
  479 | 				genericRedirectProcessor( 'giropay', transactionData, getThankYouUrl, isWhiteGloveOffer ),
  480 | 			ideal: ( transactionData ) =>
> 481 | <<<<<<< HEAD
      |  ^
  482 | 				genericRedirectProcessor( 'ideal', transactionData, getThankYouUrl, isWhiteGloveOffer ),
  483 | 			sofort: ( transactionData ) =>
  484 | 				genericRedirectProcessor( 'sofort', transactionData, getThankYouUrl, isWhiteGloveOffer ),
 ```

**Pull request merged**